packages/argo-checkout/src/extension-points/api/post-purchase/post-purchase.ts

```diff
@@ -109,3 +102,110 @@ interface Metafield {
   value: string | number;
   valueType: 'integer' | 'string' | 'json_string';
 }
+
+interface ExplicitDiscount {
+  value: number;
+  valueType: ExplicitDiscountType;
+  title: string;
+}
+
+interface AddVariantChange {
+  type: ChangeType.AddVariant;
+  variantId: number;
+  quantity: number;
+  discount?: ExplicitDiscount;
+}
+
+type Changes = AddVariantChange[];
```

Comment:
This is later going to evolve into: `type Changes = (AddVariantChange| AnotherChange | YetAnotherChange)[];`